### PR TITLE
chore: add unused rigidity env to type unification

### PR DESCRIPTION
**Pull request by mlutze**
related to #3715

### CODE REVIEW

**Comment by magnus-madsen** [comment]
@mlutze I see you added it as an argument in Typer, but I wonder if it should go in the Monad instead?

**Comment by mlutze** [comment]
> @mlutze I see you added it as an argument in Typer, but I wonder if it should go in the Monad instead?

Yes, it probably should :)

File: main/src/ca/uwaterloo/flix/language/phase/unification/Unification.scala [hunk 1]
```diff
@@ -411,18 +411,20 @@ object Unification {
     * NB: Use with EXTREME CAUTION.
     */
   def unbindVar(tvar: Type.KindedVar): InferMonad[Unit] =
-    InferMonad(s => {
-      Ok((s.unbind(tvar.sym), ()))
-    })
+    InferMonad { case (s, renv) => {
+      Ok((s.unbind(tvar.sym), renv, ()))
+    }
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/unification/Unification.scala:416]: What's with the strange indentation?

**Comment by magnus-madsen** [comment]
Blocked until we talk with Troels

File: main/src/ca/uwaterloo/flix/language/phase/unification/Unification.scala (hunk 1 above)
Comment (mlutze) [reply main/src/ca/uwaterloo/flix/language/phase/unification/Unification.scala:416]: IntelliJ failing us again :(

**Comment by magnus-madsen** [comment]
Let's do this and see how it plays 🚀

**Pull request merged**